fiftyone/utils/coco.py

```diff
@@ -2232,15 +2243,32 @@ def _instance_to_coco_segmentation(
 def _make_coco_keypoints(keypoint, frame_size):
     width, height = frame_size
 
-    # @todo true COCO format would set v = 1/2 based on whether the keypoints
-    # lie within the object's segmentation, but we'll be lazy for now
-
     keypoints = []
-    for x, y in keypoint.points:
+    num_points = len(keypoint.points)
+    visibility = [None] * num_points
+    if "visible" in keypoint:
+        if (
+            isinstance(keypoint.visible, list)
+            and len(keypoint.visible) == num_points
```

Comment:
If `keypoint.visible` is a list but the length is wrong (eg short one element etc) it might be nice to warn -- believe this will silently ignore any visibilities that _were_ supplied and fall back to the default (computing based on x,y)

Reply:
Good call! Thanks @allenleetc , I added a warning for this